### PR TITLE
Senior-id

### PR DESCRIPTION
**Pull request by kyliepace**
- adds page for users who are older than or just turned 62 to select getting a senior id
- if they select yes then they don't see the reduced fee page
- selection saved to server
- added function to calculateAge module
- rebasing lost half my changes :(

### CODE REVIEW

**Comment by kyliepace** [comment]
I tried to do git rebase origin/master but my work didn't show up in many of the merge conflict options so when I was done with the rebase much of my work was lost.

File: .vscode/settings.json [hunk 1]
```diff
@@ -0,0 +1,2 @@
+{
```
Comment (logicgraphy) [inline .vscode/settings.json]: This file should be here.

File: client/containers/apply/get-started-container.jsx [hunk 2]
```diff
@@ -2,25 +2,26 @@
...
-import navigateOnBack         from '../../helpers/navigate-on-back';
+import Form               from "../../presentations/apply/intro.jsx";
+import connectForm        from '../../helpers/connect-form';
+import navigateOnSubmit   from '../../helpers/navigate-on-submit';
+import navigateOnBack     from '../../helpers/navigate-on-back';
 
 const ConnectedForm = (props) => {
-
-  let onBack            =   props.cardType.ID === true ?  navigateOnBack('/reduced-fee', props) : navigateOnBack('/real-id', props);
+  let backAddress   =  (props.cardType.ID && props.seniorID.getSeniorID !== 'Yes') ? '/reduced-fee' : '/real-id';
```
Comment (logicgraphy) [inline client/containers/apply/get-started-container.jsx]: Back address URL is no longer required after you include changes from master. We are using history.goBack() function for back navigation and It navigates to the previously visited location.

File: client/containers/apply/real-id-container.jsx [hunk 3]
```diff
@@ -19,9 +19,13 @@ import * as dataPresent       from '../../helpers/data-present';
...
 const ConnectedForm = (props) => {
-  let address           =   props.cardType.ID ? '/reduced-fee' : '/get-started';
-  let onSubmit          =   navigateOnSubmit(address, props);
-  let onBack            =   navigateOnBack('/what-do-you-want-to-do-today', props);
+
+  //go to the get-started page unless if the ID card type has been selected and the user is not getting a senior ID
+  let nextAddress       =   (props.cardType.ID && props.seniorID.getSeniorID !== 'Yes') ? '/reduced-fee' : '/get-started';
+  let onSubmit          =   navigateOnSubmit(nextAddress, props);
+  let backAddress       =   props.seniorID.getSeniorID === 'Yes' ? '/senior-id': '/what-do-you-want-to-do-today';
+  let onBack            =   navigateOnBack(backAddress, props);
```
Comment (logicgraphy) [inline client/containers/apply/real-id-container.jsx]: backAddress no longer required.

File: client/containers/apply/senior-id-container.jsx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  const onChange = onInputChange(updateSeniorID, dispatch);
+  const onSubmit = onFormSubmit;
+
+  return {
+    onSubmit,
+    onChange
+  };
+}
+
+export default connect(mapStateToProps, mapDispatchToProps)(Form);
```
Comment (logicgraphy) [inline client/containers/apply/senior-id-container.jsx]: I think we can use connect-form.js helper here, as there is no special action to be dispatched. For eg: legal-name-form-container.jsx
Please let me know if I am missing something.

File: client/reducers/application/update-senior-id.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+import { TYPES } from '../../actions';
+import formObjectReducer from './form-object-reducer';
+
+function defaultState() {
+  return {
+    getSeniorID: ''
+  };
+}
+
+export default formObjectReducer(defaultState, TYPES.UPDATE_SENIOR_ID);
```
Comment (logicgraphy) [inline client/reducers/application/update-senior-id.js]: For single values formValueReducer could be a better choice.

**Pull request closed without merging**